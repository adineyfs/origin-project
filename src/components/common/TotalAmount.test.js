import React from "react";
import { shallow } from "enzyme";
import TestTotalAmount from "./TotalAmount";

const handleChange = jest.fn();

describe("Total Amount Component", () => {
  it("Should render with default props", () => {
    const component = shallow(<TestTotalAmount />);
    expect(component).toMatchSnapshot();
  });

  it("Should render with some value and call onChange with it", () => {
    const component = shallow(
      <TestTotalAmount amount={100.0} onChange={handleChange} />
    );
    component.find(".total-amount__input").simulate("changeEvent");
    expect(handleChange).toHaveBeenCalled();
  });
});
