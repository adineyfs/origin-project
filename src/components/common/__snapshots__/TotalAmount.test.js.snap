// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Total Amount Component Should render with default props 1`] = `
<div
  className="total-amount"
>
  <div
    className="total-amount__currency-symbol-area"
  >
    $
  </div>
  <CurrencyInput
    allowNegative={false}
    autoFocus={false}
    className="total-amount__input"
    decimalSeparator="."
    inputType="text"
    onChange={[Function]}
    onChangeEvent={[Function]}
    precision="0"
    prefix=""
    selectAllOnFocus={false}
    suffix=""
    thousandSeparator=","
    value={0}
  />
</div>
`;
